### DISCUSSION TITLE
Nova does not use Scout in Lenses

### DISCUSSION THREAD

**Discussion by dm-pf** [laravel/nova-issues#5958] [Ideas & Feature Requests]
- Laravel Version: 10.28.0
- Nova Version: 4.27.14
- PHP Version: 8.2
- Database Driver & Version:
- Operating System and Version: #.#
- Browser type and version: #.#
- Reproduction Repository: https://github.com/###/###

### Description:

I have never paid any attention how the [Scout Integration](https://nova.laravel.com/docs/search/scout-integration.html) never mentions Lenses... until today... when I received a bug report that our lenses do not return the expected results.

While investigating this - I found out that after we transitioned to Laravel Scout (which Nova says that supports out-of-the-box) - only our resources' search boxes were hitting our meilisearch instance. Lenses' search boxes have never been using Scout nor hitting our Meili.

Even though the docs never mention Scout being supported on Lenses - one would expect that to be the case - thus I'm submitting this as a "bug report" rather than a "feat request". If I'm overstepping here because of this - I apologize.

**Comment by crynobone** [comment]
> Even though the docs never mention Scout being supported on Lenses - one would expect that to be the case

Scout is only used for searching and Lens doesn't have any searching capabilities.

**Reply by dm-pf** [reply]
I know 2023 has been a pain 😞  and most of us want to stay in 2022 but we've had this for the entirety of this year. Lenses seem to rely solely on the lens having a `$search` array and doesn't care about Scout being used.

![image](https://github.com/laravel/nova-issues/assets/100700999/8fe243e2-5a20-49e8-b004-daeb90423990)

**Reply by crynobone** [reply]
That's a small subset of Laravel Nova Query Engine, Lens searching is built using `Laravel\Nova\Query\Search` while in order to make Lens support the full feature (including Scout) it has to utilize `Laraval\Nova\Query\Builder::search()` which is more complex and would require heavy rewrite in order to adapt it for Lens.

**Reply by dm-pf** [reply]
Yeah I saw that going through the code. Why weren't Lenses implemented with `Laraval\Nova\Query\Builder::search()` to begin with? And is there any dirty hack we, the users of Nova, can use today to have Lenses using Scout? Any kind of workaround would be great while waiting for this to be implemented (if ever).

**Reply by puzzledmonkey** [reply]
no offence here @crynobone but this should definitely be made a bug report - either the documentation needs updating to clearly state that lenses do NOT use scout, or the bug should be fixed to comply with nova supporting scout 'out of the box'. it's not a feature request, and it's not a discussion.

**Reply by crynobone** [reply]
Lense searching only offers what's possible using eloquent query builder, `searchableColumn()` and `$search` property is never used in Scout so this is not a simple code change. The whole API would need to be subject to changes if we want to make it possible and have potential issue with existing applications.

**Reply by puzzledmonkey** [reply]
i would think that's an internal change? if the only property exposed is a `$search` property surely the internal search implementation is up to you and can be changed without breaking any customer websites (at least ones that don't hook deeply into Nova's internals which they shouldn't be doing anyway). normal resource searching already uses scout, correct?

**Reply by crynobone** [reply]
Nope, it will be a breaking change.

**Reply by puzzledmonkey** [reply]
ok that's unfortunate, can you briefly explain why? it seems the lens and normal resource share significant methodology - both use `$search` and a standard `Builder` query (`indexQuery` for resources or `query` for lenses).

and regardless, i think it should be considered a bug report, unless the documentation specifically mentions that lenses don't use scout. otherwise the documentation is misleading and could cause confusion and wasted effort.

**Comment by crynobone** [comment]
> both use $search and a standard Builder query (indexQuery for resources or query for lenses).

No, Laravel Scout can't use `$search`, that's strictly for searching using Eloquent query builder. `query()` accepts `Illuminate\Database\Eloquent\Builder` while `Laravel\Scout\Builder` is used for Laravel Scout. That doesn't include the changes required for filters and ordering etc.
